src/improvements/doc/NEW_TEMPLATE_GUIDE.md

```diff
@@ -0,0 +1,273 @@
+
+Every template must implement the following functions:
+
+```solidity
+// Required data structures
+struct TaskConfig {
+    // Template-specific configuration parameters
+}
+
+// Storage mapping
+mapping(uint256 => TaskConfig) public taskConfig;
+
+// Core functions
+function safeAddressString() internal pure override returns (string memory);
+function _taskStorageWrites() internal pure override returns (string[] memory);
+function _templateSetup(string memory taskConfigFilePath) internal override;
+function _build(uint256 chainId) internal override;
+function _validate(uint256 chainId) internal override;
+```
+
+The struct and mapping are optional and can be customized based on task requirements. As a general rule, task developers should think of everything as a template, hence the chainId flag to the build and validate functions.
+
+### 2. Function Implementations
+
+#### safeAddressString
+Returns the name of the multisig address from the superchain-registry:
+```solidity
+function safeAddressString() internal pure override returns (string memory) {
+    return "OptimismMultisig";
+}
+```
+
+#### _taskStorageWrites
+Lists addresses from superchain-registry whose storage will be modified:
+```solidity
+function _taskStorageWrites() internal pure override returns (string[] memory) {
+    return ["SystemConfig"];
+}
+```
+
+All addresses listed here must have their storage modified during task execution.
+
+#### _templateSetup
+Initializes the task state from config.toml:
+```solidity
+function _templateSetup(string memory taskConfigFilePath) internal override {
+    // Parse config file
+    bytes memory configBytes = vm.parseToml(vm.readFile(taskConfigFilePath));
+    
+    // Save configuration to storage
+    for (uint256 i; i < l2Chains.length; i++) {
+        uint256 chainId = l2Chains[i].chainId;
+        taskConfig[chainId] = TaskConfig({
+            // Set template-specific parameters
+        });
+    }
+}
+```
+
+#### _build
+Implements the task logic for each chain:
+```solidity
+function _build(uint256 chainId) internal override {
+    TaskConfig memory config = taskConfig[chainId];
+    
+    // Implement task-specific logic
+    // Use config parameters to make state changes
+    // Handle any necessary contract interactions
+}
+```
+
+#### _validate
+Verifies the task executed correctly:
+```solidity
+function _validate(uint256 chainId) internal override {
+    TaskConfig memory config = taskConfig[chainId];
+        
+    // Verify state changes using assertEq, assertTrue, assertFalse foundry test functions
+    // make sure to user verbose error messages for better debugging
+}
+```
+
+## Configuration Structure
+
+### 1. TaskConfig Struct
+
+Define all parameters needed for the task:
+
+```solidity
+struct TaskConfig {
+    // Base parameters
+    uint256 chainId;
+    
+    // Template-specific parameters
+    uint256 gasLimit;
+    bytes32 configHash;
+    // Add other required fields
+}
+```
+
+### 2. TOML Configuration
```

Comment:
Do we need a TODO here to come back to after OPCM integration? I think the TOML file will change slightly. Also now that we're not using `.env` files. I think we want extra properties in here.